packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -31,6 +34,24 @@ abstract contract PermissionManager is Initializable {
     /// @notice A mapping storing permissions as hashes (i.e., `permissionHash(where, who, permissionId)`) and their status encoded by an address (unset, allowed, or redirecting to a `PermissionCondition`).
     mapping(bytes32 => address) internal permissionsHashed;
 
+    enum Option {
+        NONE,
+        grantOwner,
+        revokeOwner
+    }
+
+    struct Permission {
+        mapping(address => mapping(bytes32 => uint8)) delegations; // Owners can delegate the permission so delegatees can only grant it one time only.
+        mapping(address => uint8) owners;
```

Comment:
`uint8` here makes more unefficiency due to the fact that every time something is stored, first it needs to convert it from 32 bytes into 8 bytes. I think using `uint256` is ideal as we don't need packing anymore.

Reply:
okay.. yeah thought to do it explicit cause we anyways don't need that many bits